cips/cip-031.md

```diff
@@ -66,49 +65,76 @@ By integrating staking rewards directly into the lockup schedule, this proposal
+   2. The system recalculates the amount that *should* be unlocked after 30% of the period based on this new total: `Target Unlocked Amount = 0.30 * 100.657 = 30.1971 tokens`.
+   3. The difference between the target unlocked amount and the amount already unlocked is made available immediately: `Immediate Unlock = 30.1971 - 30 = 0.1971 tokens`. This represents 30% of the claimed reward (0.30 * 0.657).
+   4. The total amount now unlocked is 30.1971 tokens.
+   5. The remaining locked amount is recalculated: `New Remaining Locked = 100.657 (New Total) - 30.1971 (New Unlocked) = 70.4599 tokens`.
+   6. This new remaining locked amount (70.4599 tokens) will unlock over the remaining 70% of the period (255.5 days).
+   7. The daily unlock rate for the remainder of the period increases slightly: `New Daily Rate = 70.4599 / 255.5 ≈ 0.2758 tokens/day`.
 
-```mermaid
-sequenceDiagram
-participant DM as Distribution Module
-participant AK as Account Keeper
-participant LM as Lockup Module
-participant BM as Bank Module
+   A portion of the reward corresponding to the elapsed vesting duration (30% in this case) unlocks immediately, while the rest (70%) is added to the future vesting schedule.
 
-DM->>AK: Get Receiving Account
-AK-->>DM: Return Account
+### Delayed Vesting Accounts
 
-rect rgb(0, 0, 139)
-  Note over DM: Account Type Check
-  DM->>DM: Is Lockup Account?
-end
+   For a `DelayedVestingAccount`, the entire principal vests at a single point in the future (`EndTime`).
+   Consider an account set up with 1000 tokens scheduled to unlock entirely after 730 days (2 years).
+   Assume at day 200, the account holder claims a staking reward of 50 tokens.
 
-alt Is Lockup Account
-   DM->>LM: Get Lockup Type
-   DM->>LM: Calculate New Lockup Schedule
-   LM->>LM: Update Lockup Parameters
-   LM->>BM: Update Account Balance
-   LM-->>DM: Return Success
-else Not Lockup Account
-  DM->>BM: Process Normal Reward Transfer
-end
+   1. The reward is added to the `OriginalVesting` amount tracked by the account: `New Total Original Amount = 1000 + 50 = 1050 tokens`.
+   2. Unlike continuous vesting, no portion of this reward unlocks immediately, regardless of the time elapsed.
+   3. The account continues to hold 0 unlocked tokens.
+   4. When the `EndTime` (day 730) is reached, the *entire* 1050 tokens (original 1000 + 50 reward) unlock at once.
 
-BM-->>DM: Return Status
+   The claimed rewards simply increase the total balance that will become available at the predefined unlock date.
 
+### Periodic Vesting Accounts
 
-```
+Periodic account creation will be removed, and update schedule will be a noop
+
+### Permanent Vesting Accounts
+
+Permanent account creation will be removed, and update schedule will be a noop
+
+### Module Interaction
+
+The following sequence summarizes the module interactions:
+
+  ```mermaid
+  sequenceDiagram
+  participant DM as Distribution Module
+  participant AK as Account Keeper
+  participant LM as Lockup Module
```

Comment:
[nit] it's confusing to call this the lockup module when in code it is the vesting module

Reply:
i agree, when writing the CIP it was advised to use lockup nomenclature here even though its called vesting. I would prefer to keep it as is. on line 17 I added a note for this confusion.